rocksdb_admin/admin_handler.cpp

```diff
@@ -469,9 +469,23 @@ void AdminHandler::async_tm_addDB(
 
   // add the db to db_manager
   std::string err_msg;
+  replicator::DBRole role = replicator::DBRole::SLAVE;
+  if (request->__isset.db_role) {
+    if (request->db_role == "SLAVE") {
+      role = replicator::DBRole::SLAVE;
+    } else if (request->db_role == "MASTER") {
+      role = replicator::DBRole::MASTER;
+    } else if (request->db_role == "NOOP") {
+      role = replicator::DBRole::NOOP;
+    } else {
+      // Default behavior is to use SLAVE, so keep that
+      LOG(INFO) << "Invalid role requested to addDB " << request->db_role;
```

Comment:
Nit: it's cleaner IMO to use a switch case (but a personal preference)

Reply:
c++ doesn't support switch on a string : https://stackoverflow.com/questions/650162/why-the-switch-statement-cannot-be-applied-on-strings

I was thinking of just exposing the db role enum, but if we decide to use a bool instead we can also get rid of this